controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -236,11 +236,8 @@ public Application require(ApplicationId id) {
      * @throws IllegalArgumentException if the application already exists
      */
     public Application createApplication(ApplicationId id, Optional<NToken> token) {
-        // TODO: TLS: PR names change to prXXX.
-        if ( ! (   id.instance().value().equals("default")
-                || id.instance().value().matches("^default-pr\\d+$") // TODO: Remove when these no longer deploy.
-                || id.instance().value().matches("^\\d+$"))) // TODO: Support instances properly
-            throw new UnsupportedOperationException("Only the instance names 'default' and names starting with 'default-pr' or which are just the PR number are supported at the moment");
+        if ( ! (id.instance().value().equals("default") || id.instance().value().matches("\\d+"))) // TODO: Support instances properly
```

Comment:
Validating information about the instance should not be done here  afaik. Creating the application in Athenz should only be done once, not for each new instance.

Reply:
Is `createApplication` going to be called for each new application instance?

Reply:
It would be, with these changes, yes. That's not what we want, then?

Reply:
We do at least want validation of the new instance -- this is currently missing, apart from the application name being checked against a regular expression. As you can see, `createApplication` does a lot more than that. 

I guess we could split off the validation part of creation and just call that, and then a "lightweight" creation, when deploying new instances.

Reply:
Create application is only intended to be called before the first instance for a given application name is deployed. The instance id validation must be performed somewhere else.

Reply:
Aha. Well, when it is used that way, it is always "default". (From the create-handler.)
The problem is that we don't have a createApplication separate from createInstance, which is what this createApplication code _really_ is, and which is the reason we need the validation there. 
This will all get fixed _some day_, let's hope :)